ecs-cli/modules/cli/compose/project/project_parseV3_test.go

```diff
@@ -94,11 +137,148 @@ services:
+	expectedConfigs := getServiceConfigMap(t, project.ecsContext.ComposeFiles)
+
+	// assert # and content of container configs matches expected services
+	actualConfigs, err := project.parseV3()
 	if err != nil {
 		t.Fatal("Unexpected error parsing file: ", err)
 	}
-	assert.Equal(t, numOfServices, len(*conConfigResult))
+	assert.Equal(t, len(expectedConfigs), len(*actualConfigs))
+	for _, containerConfig := range *actualConfigs {
+		verifyConvertToContainerConfigOutput(t, expectedConfigs[containerConfig.Name], containerConfig)
+	}
+}
+
+func TestThrowErrorOnBadYaml(t *testing.T) {
+	badPortsYaml := `version: '2'
+services:
+  wordpress:
+    image: wordpress
+    ports: 
+      - "80:80", "77:77"
+  mysql:
+	image: mysql`
+
+	tmpfile, err := ioutil.TempFile("", "test")
+	if err != nil {
+		t.Fatal("Unexpected error in creating test file: ", err)
+	}
+	defer os.Remove(tmpfile.Name())
+	_, err = tmpfile.Write([]byte(badPortsYaml))
+	assert.NoError(t, err, "Unexpected error parsing file")
+
+	// add files to projects
+	project := setupTestProject(t)
+	project.ecsContext.ComposeFiles = append(project.ecsContext.ComposeFiles, tmpfile.Name())
+
+	_, err = project.parseV3()
+	assert.Error(t, err)
+}
+
+func TestThrowErrorIfFileDoesNotExist(t *testing.T) {
+	var fakeFileName = "/missingFile"
+	project := setupTestProject(t)
+	project.ecsContext.ComposeFiles = append(project.ecsContext.ComposeFiles, fakeFileName)
+	_, err := project.parseV3()
+	assert.Error(t, err)
+}
+
+func verifyConvertToContainerConfigOutput(t *testing.T, expected types.ServiceConfig, actual containerconfig.ContainerConfig) {
+
+	// verify equivalent fields
+	assert.Equal(t, expected.CapAdd, actual.CapAdd, "Expected CapAdd to match")
+	assert.Equal(t, expected.CapDrop, actual.CapDrop, "Expected CapDrop to match")
+	assert.Equal(t, expected.SecurityOpt, actual.DockerSecurityOptions, "Expected SecurityOpt and DockerSecuirtyOptions to match")
+	assert.Equal(t, []string(expected.Entrypoint), actual.Entrypoint, "Expected EntryPoint to match")
+	assert.Equal(t, expected.Name, actual.Name, "Expected Name to match")
+	assert.Equal(t, expected.Image, actual.Image, "Expected Image to match")
+	assert.Equal(t, expected.Hostname, actual.Hostname, "Expected HostName to match")
+	assert.Equal(t, expected.Links, actual.Links, "Expected Links to match")
+	assert.Equal(t, expected.Privileged, actual.Privileged, "Expected Privileged to match")
+	assert.Equal(t, expected.ReadOnly, actual.ReadOnly, "Expected ReadOnly to match")
+	assert.Equal(t, []string(expected.Command), actual.Command, "Expectd Command to match")
+	assert.Equal(t, expected.User, actual.User, "Expected User to match")
+	assert.Equal(t, expected.WorkingDir, actual.WorkingDirectory, "Expected WorkingDirectory to match")
+
+	// verify nil-able lists
+	if expected.DNSSearch != nil {
+		assert.Equal(t, []string(expected.DNSSearch), actual.DNSSearchDomains, "Expected DNSSearch and DNSSearchDomains to match")
+	} else {
```

Comment:
Apologies for missing this the first time, but in your implementation, we're not actually setting nullable fields on the containerConfig at all if `expected.<NullableField>` is nil, right? So would it make more sense to `assert.Nil` here (and below for `DNSServers`, `Tmpfs`, `Logging` etc)